### PR TITLE
Fix out of image features

### PR DESCRIPTION
**Pull request by fabiencastan**

### CODE REVIEW

File: src/openMVG/sfm/sfm_data.cpp [hunk 1]
```diff
@@ -117,7 +117,10 @@ bool ColorizeTracks( SfM_Data & sfm_data )
       if (it != obs.end())
       {
         // Color the track
-        const Vec2 & pt = it->second.x;
+        Vec2 pt = it->second.x;
+        // Clamp the pixel position if the feature/marker center is outside the image.
+        pt.x() = clamp(pt.x(), 0.0, double(image.Width()-1));
```
Comment (pmoulon) [inline src/openMVG/sfm/sfm_data.cpp]: It would be better to ensure that feature are always in the image domain before.
I mean it's an assumption.

**Comment by haavares** [comment]
I have tested this and it resolves the crash we were seeing. However, I am unable to vouch for the internal logic in OpenMvg. Ok to merge from me.

File: src/openMVG/sfm/sfm_data.cpp (hunk 1 above)
Comment (fabiencastan) [inline src/openMVG/sfm/sfm_data.cpp]: In the case of markers robust to more than 50% of occlusion (like CCTag), the feature center can be outside the image region.

Comment (pmoulon) [inline src/openMVG/sfm/sfm_data.cpp]: Ok, I better understand, and you do not want to reject those detection too early.

**Pull request merged**